core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -150,4 +152,57 @@ void shouldRecognizeInvalidGitHubUrl() throws Exception {
         assertThat(r1.status()).isEqualTo(ResultStatus.FAILURE);
         assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid");
     }
+
+    @Test
+    void shouldReturnCorrectScmFolderPath() throws IOException {
+        final Plugin p1 = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub gh = mock(GitHub.class);
+        final String repositoryName = "jenkinsci/test-repo";
+
+        Path parentDirectory = Files.createTempDirectory("jenkinsci");
```

Comment:
I have created the temp directory. But in `searchPomFiles` method  throws
`java.nio.file.NoSuchFileException: jenkinsci\test-repo` .

I do not understand why.

Reply:
Answered on Gitter but here: `Path#resolve` method is not creating anything. If you need the file or folder to be created, you need to use `Files#createX`.